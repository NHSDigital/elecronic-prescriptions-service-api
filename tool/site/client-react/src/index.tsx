import {Bundle} from "fhir/r4"
import * as React from "react"
import {
  createPrescriptionSummaryProps,
  PrescriptionSummary
} from "./components/prescription-summary/prescriptionSummary"
import {PageContainer} from "./components/pageContainer"
import ReactDOM = require("react-dom")

(async function() {
  const baseUrl = "/"
  const urlParams = new URLSearchParams(window.location.search)
  const prescriptionId = urlParams.get("prescription_id")
  const getPrescriptionResponse: Response = await fetch(`${baseUrl}prescription/${prescriptionId}`)
  const bundle: Bundle = await getPrescriptionResponse.json()
  const prescriptionSummaryProps = createPrescriptionSummaryProps(bundle)
  const content = (
    <PageContainer>
      <PrescriptionSummary
        patient={prescriptionSummaryProps.patient}
        practitionerRole={prescriptionSummaryProps.practitionerRole}
      />
    </PageContainer>
  )
  ReactDOM.render(content, document.getElementById("root"))
}())
