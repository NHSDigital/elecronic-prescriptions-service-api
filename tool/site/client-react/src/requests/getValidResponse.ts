import {AxiosResponse} from "axios"
import {OperationOutcome} from "fhir/r4"
import {UnhandledAxiosResponseError} from "./unhandledAxiosResponseError"

export function getResponseDataIfValid<T>(
  response: AxiosResponse<T | OperationOutcome>,
  typeGuard: (unknown: unknown) => unknown is T
): T {
  const responseData = response.data
  if (responseData && typeGuard(responseData)) {
    return responseData
  }

  throw new UnhandledAxiosResponseError("Unexpected response from server.", response)
}
