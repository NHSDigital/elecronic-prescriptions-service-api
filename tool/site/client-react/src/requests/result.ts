import {FhirResource} from "fhir/r4"

export function isResult(data: unknown): data is Result {
  const result = data as Result
  return typeof result.success === "boolean"
    && "request" in result
    && "request_xml" in result
    && "response" in result
    && "response_xml" in result
}

export interface Result {
  success: boolean
  request: FhirResource
  request_xml: string
  response: FhirResource
  response_xml: string
}
