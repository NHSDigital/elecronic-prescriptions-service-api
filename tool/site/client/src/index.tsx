import * as React from "react"
import {Page} from "./components/page"
import SendPreSignPage from "./pages/sendPreSignPage"
import * as ReactDOM from "react-dom"
import {BrowserRouter, Route, Switch} from "react-router-dom"
import ClaimPage from "./pages/claimPage"
import DispensePage from "./pages/dispensePage"
import {CookiesProvider} from "react-cookie"
import SendPostSignPage from "./pages/sendPostSignPage"
import HomePage from "./pages/homePage"
import PrescriptionSearchPage from "./pages/prescriptionSearchPage"
import ReleasePage from "./pages/releasePage"
import CancelPage from "./pages/cancelPage"
import MyPrescriptionsPage from "./pages/myPrescriptionsPage"
import LoginPage from "./pages/loginPage"
import ValidatePage from "./pages/validatePage"
import ReturnPage from "./pages/returnPage"
import WithdrawPage from "./pages/withdrawPage"
import LoadPage from "./pages/loadPage"
import {Environment} from "./services/environment"
import LogoutPage from "./pages/logoutPage"
import ConfigPage from "./pages/configPage"
import VerifyPage from "./pages/verifyPage"
import ComparePage from "./pages/comparePage"
import {PageFullWidth} from "./components/pageFullWidth"
import {PageContainer} from "./components/pageContainer"

const customWindow = window as Record<string, any>

export interface AppContextValue {
  baseUrl: string
  environment: Environment
}

export const AppContext = React.createContext<AppContextValue>({baseUrl: "/", environment: "int"})

async function startApplication(baseUrl: string, environment: Environment): Promise<void> {
  const urlParams = new URLSearchParams(window.location.search)
  const content = (
    <AppContext.Provider value={{baseUrl, environment}}>
      <CookiesProvider>
        <Page>
          <BrowserRouter>
            <Switch>
              <Route path={`${baseUrl}/`}>
                <PageContainer>
                  <HomePage />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}config`}>
                <PageContainer>
                  <ConfigPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}my-prescriptions`}>
                <PageContainer>
                  <MyPrescriptionsPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}login`}>
                <PageContainer>
                  <LoginPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}logout`}>
                <PageContainer>
                  <LogoutPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/load`}>
                <PageContainer>
                  <LoadPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/edit`}>
                <PageContainer>
                  <SendPreSignPage prescriptionId={urlParams.get("prescription_id")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/send`}>
                <PageContainer>
                  <SendPostSignPage token={urlParams.get("token")} state={urlParams.get("state")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/cancel`}>
                <PageContainer>
                  <CancelPage prescriptionId={urlParams.get("prescription_id")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/release`}>
                <PageContainer>
                  <ReleasePage prescriptionId={urlParams.get("prescription_id")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/verify`}>
                <PageContainer>
                  <VerifyPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/return`}>
                <PageContainer>
                  <ReturnPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/withdraw`}>
                <PageContainer>
                  <WithdrawPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/dispense`}>
                <PageContainer>
                  <DispensePage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/claim`}>
                <PageContainer>
                  <ClaimPage prescriptionId={urlParams.get("prescription_id")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}search`}>
                <PageContainer>
                  <PrescriptionSearchPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}validate`}>
                <PageContainer>
                  <ValidatePage />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}compare-prescriptions`}>
                <PageFullWidth>
                  <ComparePage />
                </PageFullWidth>
              </Route>
            </Switch>
          </BrowserRouter>
        </Page>
      </CookiesProvider>
    </AppContext.Provider>
  )
  ReactDOM.render(content, document.getElementById("root"))
}

customWindow.startApplication = startApplication
