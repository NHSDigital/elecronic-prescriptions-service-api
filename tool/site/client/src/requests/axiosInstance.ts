import axios from "axios"

export const axiosInstance = axios.create({
  validateStatus: () => true
})

// todo: refresh token if required
axiosInstance.interceptors.request.use(function (config) {
  return config
}, function (error) {
  return Promise.reject(error)
})
