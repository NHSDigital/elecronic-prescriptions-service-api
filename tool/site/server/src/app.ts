import Hapi from "@hapi/hapi"
import routes from "./routes"
import HapiPino from "hapi-pino"
import Vision from "@hapi/vision"
import * as inert from "@hapi/inert"
import Yar from "@hapi/yar"
import Cookie from "@hapi/cookie"
import {isDev, isLocal} from "./services/environment"
import axios from "axios"
import {setSessionValue} from "./services/session"

const init = async () => {
  axios.defaults.validateStatus = () => true

  const server = Hapi.server({
    port: 9000,
    host: "0.0.0.0",
    routes: {
      cors: true // Won't run as Apigee hosted target without this
    }
  })

  const baseUrl = process.env.BASE_PATH
    ? `/${process.env.BASE_PATH}/`
    : "/"

  await server.register(Cookie)
  server.auth.strategy("session", "cookie", {
    cookie: {
      name: "auth",
      password: process.env.SESSION_TOKEN_ENCRYPTION_KEY,
      isSecure: true
    },
    redirectTo: (request: Hapi.Request) => {
      if (isDev()) {
        setSessionValue(
          "use_signing_mock",
          request.query["use_signing_mock"],
          request
        )
      }
      return `${baseUrl}login`
    }
  })
  server.auth.default("session")

  server.route(routes)

  await server.register({
    plugin: Yar,
    options: {
      storeBlank: false,
      // Use "0" maxCookieSize to force all session data to be written to cache
      maxCookieSize: 0,
      cache: {
        expiresIn: 24 * 60 * 60 * 1000
      },
      cookieOptions: {
        password: process.env.SESSION_TOKEN_ENCRYPTION_KEY ?? "",
        isSecure: true,
        isSameSite: "None"
      }
    }
  })

  await server.register({
    plugin: HapiPino,
    options: {
      // Pretty print in local environment only to avoid spamming logs
      prettyPrint: isLocal(),
      // Redact Authorization headers, see https://getpino.io/#/docs/redaction
      redact: ["req.headers.authorization"]
    }
  })

  await server.register(inert)

  server.route({
    method: "GET",
    path: `/static/{param*}`,
    options: {
      auth: false
    },
    handler: {
      directory: {
        path: "static"
      }
    }
  })

  await server.register(Vision)

  server.views({
    engines: {
      html: require("handlebars")
    },
    relativeTo: __dirname,
    path: "templates"
  })

  server.route(addHomeViewRoute())
  server.route(addViewRoute("login", true))
  server.route(addViewRoute("my-prescriptions"))
  server.route(addViewRoute("validate"))
  server.route(addViewRoute("search"))
  server.route(addViewRoute("prescribe/load"))
  server.route(addViewRoute("prescribe/edit"))
  server.route(addViewRoute("prescribe/send"))
  server.route(addViewRoute("prescribe/cancel"))
  server.route(addViewRoute("dispense/release"))
  server.route(addViewRoute("dispense/return"))
  server.route(addViewRoute("dispense/dispense"))
  server.route(addViewRoute("dispense/withdraw"))
  server.route(addViewRoute("dispense/claim"))

  function addHomeViewRoute() : Hapi.ServerRoute {

    const baseUrl = process.env.BASE_PATH
      ? `/${process.env.BASE_PATH}/`
      : "/"

    return {
      method: "GET",
      path: `/`,
      handler: {
        view: {
          template: "index",
          context: {
            baseUrl,
            environment: process.env.ENVIRONMENT
          }
        }
      }
    }
  }

  function addViewRoute(path: string, skipAuth?: boolean): Hapi.ServerRoute {
    const baseUrl = process.env.BASE_PATH
      ? `/${process.env.BASE_PATH}/`
      : "/"

    const viewRoute = {
      method: "GET",
      path: `/${path}`,
      handler: {
        view: {
          template: "index",
          context: {
            baseUrl,
            environment: process.env.ENVIRONMENT
          }
        }
      }
    }

    if (skipAuth) {
      return {
        ...viewRoute,
        options: {
          auth: false
        }
      }
    }

    return viewRoute
  }

  await server.start()
  server.log("info", `Server running on ${server.info.uri}`)
}

process.on("unhandledRejection", err => {
  console.log(err)
  process.exit(1)
})

init()
