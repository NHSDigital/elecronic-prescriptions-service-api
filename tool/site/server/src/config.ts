import {Environment} from "./services/environment"

export interface Config {
  commitId: string
  validatorVersion: string
  basePath: string
  baseUrl: string
  environment: Environment
  sessionKey: string
  privateApigeeUrl: string
  publicApigeeUrl: string
  clientId: string
  clientSecret: string
  privateKey: string
  keyId: string
  subject: string
  cis2Secret: string
  cis2EgressHost: string
  refreshTokenTimeout: number
}

export const CONFIG: Config = {
  commitId: process.env.COMMIT_ID ?? "unknown",
  validatorVersion: process.env.VALIDATOR_VERSION ?? "unknown",
  basePath: process.env.BASE_PATH ?? "eps-api-tool",
  baseUrl: process.env.BASE_PATH ? `/${process.env.BASE_PATH}/` : "/",
  environment: process.env.ENVIRONMENT as Environment ?? "int",
  sessionKey: process.env.SESSION_TOKEN_ENCRYPTION_KEY ?? "",
  privateApigeeUrl: `https://${process.env.APIGEE_DOMAIN_NAME}`,
  publicApigeeUrl: process.env.PUBLIC_APIGEE_URL ?? "",
  clientId: process.env.DEMO_APP_CLIENT_ID ?? "",
  clientSecret: process.env.DEMO_APP_CLIENT_KEY ?? "",
  privateKey: process.env.DEMO_APP_PRIVATE_KEY ?? "",
  keyId: process.env.DEMO_APP_KEY_ID ?? "",
  subject: process.env.APP_JWT_SUBJECT ?? "",
  cis2Secret: process.env.DEMO_APP_CIS2_KEY ?? "",
  cis2EgressHost: process.env.CIS2_EGRESS_HOST ?? "",
  // full refresh timeout seconds is divided by 3 as only one refresh is working atm
  // times by 1000 to get miliseconds as ttls for cookies takes ms
  refreshTokenTimeout: (3599 / 3) * 1000
}
