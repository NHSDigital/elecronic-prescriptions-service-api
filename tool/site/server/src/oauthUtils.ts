import ClientOAuth2 from "client-oauth2"
import {URL} from "url"
import {CONFIG} from "./config"
import {getRegisteredCallbackUrl} from "./routes/helpers"

export interface OAuthClient {
  getUri: (options?: ClientOAuth2.Options) => string
  getToken: (uri: URL) => Promise<Token>
}

export interface Token {
  data: ClientOAuth2.Data
  tokenType: string
  accessToken: string
  refreshToken: string
}

const oauthClient =
  new ClientOAuth2({
    clientId: CONFIG.clientId,
    clientSecret: CONFIG.clientSecret,
    redirectUri: getRegisteredCallbackUrl("callback"),
    accessTokenUri: `${CONFIG.privateApigeeUrl}/oauth2/token`,
    authorizationUri: `${CONFIG.publicApigeeUrl}/oauth2/authorize`,
    body: {
      client_id: CONFIG.clientId,
      client_secret: CONFIG.clientSecret
    }
  })

export default function getOAuthClient(): OAuthClient {
  return oauthClient.code
}

export async function refreshToken(data: ClientOAuth2.Data): Promise<ClientOAuth2.Token> {
  const oauthClientToken = oauthClient.createToken(data)
  const refreshedToken = await oauthClientToken.refresh()
  oauthClientToken.expiresIn(parseInt(oauthClientToken.data.refreshTokenExpiresIn))
  return refreshedToken
}
