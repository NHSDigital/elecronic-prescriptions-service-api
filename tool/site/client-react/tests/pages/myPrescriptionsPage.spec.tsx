import {waitFor} from "@testing-library/react"
import {screen} from "@testing-library/dom"
import pretty from "pretty"
import * as React from "react"
import moxios from "moxios"
import {AppContextValue} from "../../src"
import {renderWithContext} from "../renderWithContext"
import {axiosInstance} from "../../src/requests/axiosInstance"
import MyPrescriptionsPage from "../../src/pages/myPrescriptionsPage"

const baseUrl = "baseUrl/"
const context: AppContextValue = {baseUrl}

const prescriptionsUrl = `${baseUrl}prescriptions`

beforeEach(() => moxios.install(axiosInstance))

afterEach(() => moxios.uninstall(axiosInstance))

test("Displays my prescriptions page", async () => {
  moxios.stubRequest(prescriptionsUrl, {
    status: 200,
    response: {
      any: false,
      sentPrescriptions: [],
      releasedPrescriptions: []
    }
  })

  const container = await renderPage()

  expect(screen.getByText("My Prescriptions")).toBeTruthy()
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

test("Displays sent prescriptions from session", async () => {
  moxios.stubRequest(prescriptionsUrl, {
    status: 200,
    response: {
      any: true,
      sentPrescriptions: [{id: "FC6D78-A83008-EDF7BI"}],
      releasedPrescriptions: []
    }
  })

  const container = await renderPage()
  await waitFor(() => screen.getByText(/Sent Prescriptions/))
  expect(screen.getByText("FC6D78-A83008-EDF7BI")).toBeTruthy()
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

test("Displays released prescriptions from session", async () => {
  moxios.stubRequest(prescriptionsUrl, {
    status: 200,
    response: {
      any: true,
      sentPrescriptions: [],
      releasedPrescriptions: [{id: "FC6D78-A83008-EDF7BF"}]
    }
  })

  const container = await renderPage()
  await waitFor(() => screen.getByText(/Released Prescriptions/))
  expect(screen.getByText("FC6D78-A83008-EDF7BF")).toBeTruthy()
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

async function renderPage() {
  const {container} = renderWithContext(<MyPrescriptionsPage/>, context)
  await waitFor(() => screen.getByText("My Prescriptions"))
  return container
}
