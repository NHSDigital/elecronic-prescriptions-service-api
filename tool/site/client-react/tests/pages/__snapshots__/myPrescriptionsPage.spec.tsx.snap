// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displays my prescriptions page 1`] = `
"<h1 class=\\"nhsuk-label-wrapper\\"><label class=\\"nhsuk-label nhsuk-label--xl\\">My Prescriptions</label></h1>
<p>You do not have any active prescriptions. Once you create or release a prescription you will see it here</p>"
`;

exports[`Displays released prescriptions from session 1`] = `
"<h1 class=\\"nhsuk-label-wrapper\\"><label class=\\"nhsuk-label nhsuk-label--xl\\">My Prescriptions</label></h1>
<div class=\\"nhsuk-table__panel-with-heading-tab\\">
  <h3 class=\\"nhsuk-table__heading-tab\\">Released Prescriptions</h3>
  <div class=\\"nhsuk-table-responsive\\">
    <table class=\\"nhsuk-table\\">
      <caption class=\\"nhsuk-table__caption\\">Prescriptions ready to dispense</caption>
      <thead class=\\"nhsuk-table__head\\">
        <tr class=\\"nhsuk-table__row\\">
          <th class=\\"nhsuk-table__header\\" scope=\\"col\\">ID</th>
          <th class=\\"nhsuk-table__header\\" scope=\\"col\\">Actions</th>
        </tr>
      </thead>
      <tbody class=\\"nhsuk-table__body\\">
        <tr class=\\"nhsuk-table__row\\">
          <td class=\\"nhsuk-table__cell\\">FC6D78-A83008-EDF7BF</td>
          <td class=\\"nhsuk-table__cell\\">
            <div class=\\"nhsuk-action-link\\"><a class=\\"nhsuk-action-link__link\\" href=\\"baseUrl/dispense/return?prescription_id=FC6D78-A83008-EDF7BF\\"><svg class=\\"nhsuk-icon nhsuk-icon__arrow-right-circle\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"32\\" width=\\"32\\">
                  <path d=\\"M0 0h24v24H0z\\" fill=\\"none\\"></path>
                  <path d=\\"M12 2a10 10 0 0 0-9.95 9h11.64L9.74 7.05a1 1 0 0 1 1.41-1.41l5.66 5.65a1 1 0 0 1 0 1.42l-5.66 5.65a1 1 0 0 1-1.41 0 1 1 0 0 1 0-1.41L13.69 13H2.05A10 10 0 1 0 12 2z\\"></path>
                </svg><span class=\\"nhsuk-action-link__text\\">Return prescription</span></a></div>
            <div class=\\"nhsuk-action-link\\"><a class=\\"nhsuk-action-link__link\\" href=\\"baseUrl/dispense/dispense?prescription_id=FC6D78-A83008-EDF7BF\\"><svg class=\\"nhsuk-icon nhsuk-icon__arrow-right-circle\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"32\\" width=\\"32\\">
                  <path d=\\"M0 0h24v24H0z\\" fill=\\"none\\"></path>
                  <path d=\\"M12 2a10 10 0 0 0-9.95 9h11.64L9.74 7.05a1 1 0 0 1 1.41-1.41l5.66 5.65a1 1 0 0 1 0 1.42l-5.66 5.65a1 1 0 0 1-1.41 0 1 1 0 0 1 0-1.41L13.69 13H2.05A10 10 0 1 0 12 2z\\"></path>
                </svg><span class=\\"nhsuk-action-link__text\\">Dispense prescription</span></a></div>
            <div class=\\"nhsuk-action-link\\"><a class=\\"nhsuk-action-link__link\\" href=\\"baseUrl/search?prescription_id=FC6D78-A83008-EDF7BF\\"><svg class=\\"nhsuk-icon nhsuk-icon__arrow-right-circle\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"32\\" width=\\"32\\">
                  <path d=\\"M0 0h24v24H0z\\" fill=\\"none\\"></path>
                  <path d=\\"M12 2a10 10 0 0 0-9.95 9h11.64L9.74 7.05a1 1 0 0 1 1.41-1.41l5.66 5.65a1 1 0 0 1 0 1.42l-5.66 5.65a1 1 0 0 1-1.41 0 1 1 0 0 1 0-1.41L13.69 13H2.05A10 10 0 1 0 12 2z\\"></path>
                </svg><span class=\\"nhsuk-action-link__text\\">View prescription</span></a></div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>"
`;

exports[`Displays sent prescriptions from session 1`] = `
"<h1 class=\\"nhsuk-label-wrapper\\"><label class=\\"nhsuk-label nhsuk-label--xl\\">My Prescriptions</label></h1>
<div class=\\"nhsuk-table__panel-with-heading-tab\\">
  <h3 class=\\"nhsuk-table__heading-tab\\">Sent Prescriptions</h3>
  <div class=\\"nhsuk-table-responsive\\">
    <table class=\\"nhsuk-table\\">
      <caption class=\\"nhsuk-table__caption\\">Prescriptions ready to release</caption>
      <thead class=\\"nhsuk-table__head\\">
        <tr class=\\"nhsuk-table__row\\">
          <th class=\\"nhsuk-table__header\\" scope=\\"col\\">ID</th>
          <th class=\\"nhsuk-table__header\\" scope=\\"col\\">Actions</th>
        </tr>
      </thead>
      <tbody class=\\"nhsuk-table__body\\">
        <tr class=\\"nhsuk-table__row\\">
          <td class=\\"nhsuk-table__cell\\">FC6D78-A83008-EDF7BI</td>
          <td class=\\"nhsuk-table__cell\\">
            <div class=\\"nhsuk-action-link\\"><a class=\\"nhsuk-action-link__link\\" href=\\"baseUrl/dispense/release?prescription_id=FC6D78-A83008-EDF7BI\\"><svg class=\\"nhsuk-icon nhsuk-icon__arrow-right-circle\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"32\\" width=\\"32\\">
                  <path d=\\"M0 0h24v24H0z\\" fill=\\"none\\"></path>
                  <path d=\\"M12 2a10 10 0 0 0-9.95 9h11.64L9.74 7.05a1 1 0 0 1 1.41-1.41l5.66 5.65a1 1 0 0 1 0 1.42l-5.66 5.65a1 1 0 0 1-1.41 0 1 1 0 0 1 0-1.41L13.69 13H2.05A10 10 0 1 0 12 2z\\"></path>
                </svg><span class=\\"nhsuk-action-link__text\\">Release prescription</span></a></div>
            <div class=\\"nhsuk-action-link\\"><a class=\\"nhsuk-action-link__link\\" href=\\"baseUrl/search?prescription_id=FC6D78-A83008-EDF7BI\\"><svg class=\\"nhsuk-icon nhsuk-icon__arrow-right-circle\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"32\\" width=\\"32\\">
                  <path d=\\"M0 0h24v24H0z\\" fill=\\"none\\"></path>
                  <path d=\\"M12 2a10 10 0 0 0-9.95 9h11.64L9.74 7.05a1 1 0 0 1 1.41-1.41l5.66 5.65a1 1 0 0 1 0 1.42l-5.66 5.65a1 1 0 0 1-1.41 0 1 1 0 0 1 0-1.41L13.69 13H2.05A10 10 0 1 0 12 2z\\"></path>
                </svg><span class=\\"nhsuk-action-link__text\\">View prescription</span></a></div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>"
`;
